lib/configuration.rb

```diff
@@ -109,4 +109,55 @@ def get_optional_class_instance(keyword)
       return Object.const_get(class_opts[:class]).new(**class_opts[:opts])
     end
   end
+
+  class FlexyEnvVariable
+    attr_reader :var_name
+
+    def initialize(var_name)
+      @var_name=var_name
+    end
+    def ==(other)
+      self.class===other &&
+        @var_name==other.var_name
+    end
+    def to_s
+      ENV[@var_name]
+    end
+    
+    alias :eql? :==
+    alias :to_str :to_s
+    alias :[]= :to_s
+  end
+
+  class FlexyEnvFile
+    @@files = Hash.new
+    attr_reader :var_name
+
+    def initialize(var_name)
+      @var_name=var_name
+    end
+    def ==(other)
+      self.class===other && 
+        @var_name==other.var_name
+    end
+    def to_s
+      file=@@files[@var_name]
+      if !file.nil? && !file.file?
+        @@files.except!(@var_name)
+        file=nil
+      end
```

Comment:
Could you give an example where `if` condition above is needed. How can file be not `nil` and not a file? btw if `file` is not a file, then it probably doesn't have the method `file?` in which case an exception will be raised. Again, you might be right, I might be missing the point.

Reply:
My intention was to check where I retrieve here a stale `File` handle or not. I know the POSIX behavior allows to have a file opened and call unlink - filesystem doesn't report that file existence onward but handle owner process can still call any operation over that file... 

On the other hand we have an opposite situation - we have a `Tempfile`, it was closed (after exiting the resource block) however GC can't delete it because there isn't a reference counter lowered to zero (intentionally, we put it into the map just for this intentional case). On the other hand, we have a handle for closed file stored then - this file can be deleted externally (directly on FS - it was closed, so I don't expect any claims here) and in that case I believe we cat get into the situation when we have stale  file handler (file.nil -> false) but the file doesn't exists (file.file? -> false). In this case I drop the reference and going forward like the first get call.

Reply:
If I'm not wrong (see https://ruby-doc.org/core-2.5.0/File.html#method-c-file-3F) - "Returns true if the named file exists and is a regular file." Which perfectly fits to our case or am I missing sth here?

Reply:
Got it. I wouldn't bother to test for such an edge case but it is a possible situation. If you like to keep this check it may help adding a comment line like `# file has been deleted on filesystem`.

Reply:
:+1:

Reply:
Verified, these aliases aren't necessary now thus removing them. It had to be an issue on my side when coding then...